dds/DCPS/FilterEvaluator.cpp

```diff
@@ -342,21 +338,63 @@ namespace {
 
   private:
     bool invert_;
-    FieldLookup* field_;
-    Operand* left_;
-    Operand* right_;
+    FilterEvaluator::Operand* field_;
+    FilterEvaluator::Operand* left_;
+    FilterEvaluator::Operand* right_;
+  };
+
+  class Call : public FilterEvaluator::Operand {
+  public:
+    enum Operator { OP_MOD };
+
+    Call(const OPENDDS_STRING& name)
+    {
+      if (name == "MOD") {
+        op_ = OP_MOD;
+      } else {
+        assert(0);
+      }
+    }
+
+    virtual Value eval(FilterEvaluator::DataForEval& data)
+    {
+      switch (op_) {
+      case OP_MOD:
+        {
+          assert(children_.size() == 2);
```

Comment:
Once the grammar is updated, make this throw an exception (changing from syntax error to semantic error).